lib/ansible/modules/cloud/opennebula/one_host.py

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/python
+#
+# Copyright 2018 www.privaz.io Valletech AB
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: one_host
+
+short_description: Manages OpenNebula Hosts
+
+version_added: "2.6"
+
+requirements:
+    - pyone
+
+description:
+    - "Manages OpenNebula Hosts"
+
+options:
+    name:
+        description:
+            - Hostname of the machine to manage.
+        required: true
+    state:
+        description:
+            - Takes the host to the desired lifecycle state.
+            - If C(absent) the host will be deleted from the cluster.
+            - If C(present) the host will be created in the cluster (includes C(enabled), C(disabled) and C(offline) states).
+            - If C(enabled) the host is fully operational.
+            - C(disabled), e.g. to perform maintenance operations.
+            - C(offline), host is totally offline.
+        choices:
+            - absent
+            - present
+            - enabled
+            - disabled
+            - offline
+        default: present
+    im_mad_name:
+        description:
+            - The name of the information manager, this values are taken from the oned.conf with the tag name IM_MAD (name)
+        default: kvm
+    vmm_mad_name:
+        description:
+            - The name of the virtual machine manager mad name, this values are taken from the oned.conf with the tag name VM_MAD (name)
+        default: kvm
+    cluster_id:
+        description:
+            - The cluster ID.
+        default: 0
+    cluster_name:
+        description:
+            - The cluster specified by name.
+    template:
+        description:
+            - The template changes to merge into the host template.
+
+extends_documentation_fragment: opennebula
+
+author:
+    - Rafael del Valle (@rvalle)
+'''
+
+EXAMPLES = '''
+- name: Create a new host in OpenNebula
+  one_host:
+    name: host1
+    cluster_id: 1
+    endpoint: http://127.0.0.1:2633/RPC2
+
+- name: Create a host and adjust its template
+  one_host:
+    name: host2
+    cluster_name: default
+    template:
+        LABELS:
+            - gold
+            - ssd
+        RESERVED_CPU: -100
+'''
+
+# TODO: what makes most sense to return?
```

Comment:
I am not yet returning any values. And I have left an open point about it to make a decision later in the design guidelines.

Reply:
It would be nice to return at least a list of instances ids deployed on that host, so we can connect it with `one_vm`

Reply:
I believe you are after the flush operation. When you ask a to re-schedule all VMs because you are going to put the host for maintenance? is that the use case?
we can discuss in call next week.

Reply:
`one_host` module shouldn't return IDs of instances running on it, because this module doesn't work with VMs. Returning a list of all VMs on some host should be part of the `one_vm` module, but we will discuss this next week.